tests/Routing/RoutingUrlGeneratorTest.php

```diff
@@ -115,6 +115,64 @@ public function testBasicRouteGeneration()
 	}
 
 
+	public function testBasicRouteGenerationOptionalParameterMatch()
+	{
+		$url = new UrlGenerator(
+				$routes = new Illuminate\Routing\RouteCollection,
+				$request = Illuminate\Http\Request::create('http://www.foo.com/')
+		);
+
+		$route = new Illuminate\Routing\Route(array('GET'), 'foo/bar/{baz?}', array('as' => 'foobaz'));
+		$routes->add($route);
+
+		$this->assertEquals('http://www.foo.com/foo/bar/taylor', $url->route('foobaz', ['baz' => 'taylor']));
+	}
+
+
+	public function testBasicRouteGenerationOptionalParameterNoMatch()
+	{
+		$url = new UrlGenerator(
+			$routes = new Illuminate\Routing\RouteCollection,
+			$request = Illuminate\Http\Request::create('http://www.foo.com/')
+		);
+
+		$route = new Illuminate\Routing\Route(array('GET'), 'foo/bar/{baz?}', array('as' => 'foobaz'));
+		$routes->add($route);
+
+		$this->assertEquals('http://www.foo.com/foo/bar?taylor=otwell', $url->route('foobaz', ['taylor' => 'otwell']));
+	}
+
+
+	public function testBasicRouteGenerationOptionalParameterNoMatchIntegerKeyValue()
+	{
+		$url = new UrlGenerator(
+				$routes = new Illuminate\Routing\RouteCollection,
+				$request = Illuminate\Http\Request::create('http://www.foo.com/')
+		);
+
+		$route = new Illuminate\Routing\Route(array('GET'), 'foo/bar/{baz?}', array('as' => 'foobaz'));
+		$routes->add($route);
+
+		// or /foo/bar?1&2 ???
+		$this->assertEquals('http://www.foo.com/foo/bar//1/2', $url->route('foobaz', [1,2]));
```

Comment:
The UrlGenerator class has different public methods, at least action() and route(). Is this a possibility to implement different algorithms? If not, any other thoughts how this can be fixed?